src/lib.rs

```diff
@@ -1,3 +1,4 @@
+pub mod authentication;
```

Comment:
Pulled all of the validate credentials logic into a separate module since the login route and newsletter route will be using it